sqlglot/parser.py

```diff
@@ -1933,7 +1939,7 @@ def _parse_function(self, functions=None):
             else:
                 this = self.expression(exp.Anonymous, this=this, expressions=args)
         self._match_r_paren()
-        return self._parse_window(this)
+        return self._parse_window(self._attach_comment(this, self._prev.comment))
```

Comment:
instead of wrapping _attach_comment everywhere, what if you just added it to

def expression ? so anytime we make an expression we also parse for comments

Reply:
That sounds interesting, although I'm concerned it might not take care of lower level calls that don't delegate to `parse_expression` in their body. I'll try it.